pkg/backup/rbac.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backup
+
+import (
+	arkdiscovery "github.com/heptio/ark/pkg/discovery"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	rbac "k8s.io/api/rbac/v1"
+	rbacbeta "k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	rbacclient "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	rbacclientbeta "k8s.io/client-go/kubernetes/typed/rbac/v1beta1"
+)
+
+type CRBLister interface {
+	// List returns a slice of ClusterRoleBindings which can represent either v1 or v1beta1 ClusterRoleBindings.
+	List() ([]ClusterRoleBinding, error)
+}
+
+type V1CRBLister struct {
+	client rbacclient.ClusterRoleBindingInterface
+}
+
+func (v1 V1CRBLister) List() ([]ClusterRoleBinding, error) {
+	var crbs []ClusterRoleBinding
+
+	crbList, err := v1.client.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	for _, crb := range crbList.Items {
+		crbs = append(crbs, v1crb{crb: crb})
+	}
+
+	return crbs, nil
+}
+
+type V1beta1CRBLister struct {
+	client rbacclientbeta.ClusterRoleBindingInterface
+}
+
+func (v1beta1 V1beta1CRBLister) List() ([]ClusterRoleBinding, error) {
+	var crbs []ClusterRoleBinding
```

Comment:
Wanted to extend on what @skriss said to make sure this and I think one additional case of this is also changed: I'd move this `crbs` var to right above the `for` loop where it's used. It's a common thing in Go so it's expected but also avoids scrolling all over for where it's used if we expect its first use to happen right after where it's declared.